server/admin-function/check-admin.ts

```diff
@@ -0,0 +1,21 @@
+"use server";
+
+import { dbAdmin } from "@/lib/firebase-admin";
+
+export const checkAdmin = async (domain: string, loggedInUserId:string): Promise<boolean> => {
+  try {
+
+    console.log('checkAdmin:', domain, loggedInUserId);
+    const userDocRef = dbAdmin.doc(`domain/${domain}/users/${loggedInUserId}`);
+    const userDocSnap = await userDocRef.get();
+    const userRole = userDocSnap.data()?.role;
```

Comment:
again, would their be a way to avoid fetching data again here everything we check admin, we're fetching the same multiple times for all different tasks

maybe a server side cache with encrypted data or something like that could avoid this amount of repetitive fetch

Reply:
this isnt use yet, this will be used later to verify if the user is admin before fetch or writing things only admin can do